src/core/config/config_test.go

```diff
@@ -509,19 +512,26 @@ func TestDeprecatedEnvPrefixMigration(t *testing.T) {
 	assert.Equal(t, want, got)
 }
 
-func setEnvVariable(t *testing.T, name string, value string) {
-	key := strings.ToUpper(EnvPrefix + agent_config.KeyDelimiter + name)
-	err := os.Setenv(key, value)
+func setEnvVariable(t *testing.T, prefix, name, value string) {
+	var err error
+	key := strings.ToUpper(prefix + agent_config.KeyDelimiter + name)
+	if value == "" {
+		err = os.Unsetenv(key)
+	} else {
+		err = os.Setenv(key, value)
```

Comment:
Could we use `t.Setenv()` instead of `os.Setenv()` here? The `t` version is automatically cleaned up.